### DISCUSSION TITLE
make a new string-specific validations isLowerCase  &  isUpperCase

### DISCUSSION THREAD

**Discussion by sayandedotcom** [colinhacks/zod#3183] [Q&A] [answered]
I want to check whether the user input  "username" is in lowercase with Zod..but I couldn't find anything like isLowerCase, isUpperCase .

**Comment by JacobWeisenburger** [comment]
Is there a reason you need to know if something is lower case? Because you can just as easily change it to lower case without ever needing to check it.

**Comment by saidMounaim** [comment]
check this, for lowercase, if u want uppercase just change str.toLowerCase() to str.toUpperCase()

`name: z.string().min(1, "name is required").max(100).refine((str: string) => str === str.toLowerCase(), "Name must be Lowercase")`

**Comment by JacobWeisenburger** [comment, marked as answer]
No need to check if it's lowercase, just make it lowercase.

```ts
const schema = z.object( { username: z.string().toLowerCase() } )

console.log( schema.parse( { username: 'Foo' } ) ) // { username: 'foo' }
console.log( schema.parse( { username: 'FOO' } ) ) // { username: 'foo' }
console.log( schema.parse( { username: 'FoO' } ) ) // { username: 'foo' }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger
